packages/medusa/src/api-v2/store/customers/route.ts

```diff
@@ -1,14 +1,37 @@
 import { MedusaRequest, MedusaResponse } from "../../../types/routing"
-import { createCustomerAccountWorkflow } from "@medusajs/core-flows"
+
+import { ContainerRegistrationKeys } from "@medusajs/utils"
 import { CreateCustomerDTO } from "@medusajs/types"
+import { createCustomerAccountWorkflow } from "@medusajs/core-flows"
 
 export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  if (req.auth_user?.app_metadata?.customer_id) {
+    const remoteQuery = req.scope.resolve(
+      ContainerRegistrationKeys.REMOTE_QUERY
+    )
+    const query = {
```

Comment:
suggestion: should we start using `remoteQueryObjectFromString` everywhere to normalize and that way we don't have to change everywhere when we change the property names

```ts
const query = remoteQueryObjectFromString({ entryPoint: "customer", variables: { id: ... }, fields: [])
```